pkg/network/network_test.go

```diff
@@ -490,6 +490,121 @@ func TestConfiguration(t *testing.T) {
 	}
 }
 
+// TODO move this to TestConfiguration
+func TestSubDomain(t *testing.T) {
+	subdomainTemplate := `{{if eq (index .Annotations "sub") ""}}{{.Name}}.{{.Namespace}}.{{.Domain}}{{else}}{{.Name}}.{{ index .Annotations "sub"}}.{{.Domain}}{{end}}`
+	networkConfigTests := []struct {
+		name       string
+		wantErr    bool
+		wantConfig *Config
+		config     *corev1.ConfigMap
+		data       DomainTemplateValues
+	}{{
+		name:    "network configuration with annotations in template",
+		wantErr: false,
+		wantConfig: &Config{
+			IstioOutboundIPRanges:      "*",
+			DefaultClusterIngressClass: "foo-ingress",
+			DomainTemplate:             subdomainTemplate,
```

Comment:
Should this be a hardcoded value so we can verify that the templating worked?

Reply:
done